### PR TITLE
feat(JSONTree): allow case insensitive search

### LINKED ISSUE #1567: Make search in describe case-insensitive
image

image

### PR DESCRIPTION
**Pull request by Raubzeug**
Closes #1567

[Stand](https://nda.ya.ru/t/-nqWFU4g7AAjfz)

### Test Status: <span style="color: green;">✅ PASSED</span>
  📊 [Full Report](https://ydb-platform.github.io/ydb-embedded-ui/1735/)

  | Total | Passed | Failed | Flaky | Skipped |
  |:-----:|:------:|:------:|:-----:|:-------:|
  | 208 | 208 | 0 | 0 | 0 |

  😟 No changes in tests. 😕

  ### Bundle Size: ✅
  Current: 65.97 MB | Main: 65.97 MB
  Diff: +3.00 KB (0.00%)

  ✅ Bundle size unchanged.

  <details>
  <summary>ℹ️ CI Information</summary>

  - Test recordings for failed tests are available in the full report.
  - Bundle size is measured for the entire 'dist' directory.
  - 📊 indicates links to detailed reports.
  - 🔺 indicates increase, 🔽 decrease, and ✅ no change in bundle size.
  </details>

### CODE REVIEW

File: src/components/JSONTreeWithSearch/JSONTreeWithSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    );
+    return (
+        <div className={b()}>
+            <JSONTree
+                className={b('tree', treeClassName)}
+                filterOptions={{
+                    ignoreCase: !caseSensitiveSearch,
+                }}
+                searchOptions={{
+                    debounceTime: 300,
```
Comment (astandrik) [inline src/components/JSONTreeWithSearch/JSONTreeWithSearch.tsx:34]: const DEBOUNCE_TIME ?

File: src/services/settings.ts [hunk 2]
```diff
@@ -46,6 +47,7 @@ export const DEFAULT_USER_SETTINGS = {
     [AUTOCOMPLETE_ON_ENTER]: true,
     [IS_HOTKEYS_HELP_HIDDEN_KEY]: false,
     [AUTO_REFRESH_INTERVAL]: 0,
+    [CASE_SENSITIVE_JSON_SEARCH]: false,
```
Comment (astandrik) [inline src/services/settings.ts:50]: looks like global option but  as far as I see JSONTree used in many places

image

Comment (astandrik) [reply src/services/settings.ts:50]: I suppose case-sensitiveness should be scoped to JSONTreeWrapper (or JSONTreeSearch or whatever component it is used in)

Comment (Raubzeug) [reply src/services/settings.ts:50]: It's a setting that should be saved to LS. Thats why it is among other settings. Or I didn't get you?..

**Review by astandrik — APPROVED** [review]

**Pull request merged**